node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -269,28 +295,38 @@ func (ms *P2PMessageService) Send(msg protocols.Message) {
 	raw, err := msg.Serialize()
 	ms.checkError(err)
 
+	// First try to get peerId from local "peers" map. If the address is not found there,
+	// query the dht to retrieve the peerId, then store in local map for next time
 	peerId, ok := ms.peers.Load(msg.To.String())
 	if !ok {
-		panic(fmt.Errorf("could not load peer %s", msg.To.String()))
+		ms.logger.Info().Msgf("did not find address %s in local map, will query dht", msg.To.String())
```

Comment:
This logic could also be in it's own function, that could help reduce the overall complexity of `Send`